src/main/kotlin/com/exactpro/th2/codec/EncodeProcessor.kt

```diff
@@ -50,8 +51,16 @@ class EncodeProcessor(
                 continue
             }
 
-            if (!messageGroup.isEncodable()) {
-                parentEventId.onErrorEvent("No messages of $protocols protocol or mixed empty and non-empty protocols are present", messageGroup.messageIds)
+            val msgProtocols = messageGroup.protocols
+            if (msgProtocols.none(String::isBlank) && protocols.none { it in msgProtocols }) {
+                logger.debug { "Messages with $msgProtocols protocols instead of $protocols are presented" }
+                messageBatch.addGroups(messageGroup)
+                continue
+            } else if (msgProtocols.all(String::isBlank) || msgProtocols.none(String::isBlank) && protocols.any { it in msgProtocols }) {
+                // do nothing
+            } else {
+                val info = "No messages of $protocols protocols or mixed empty and non-empty protocols are present"
+                parentEventId.onErrorEvent(info, messageGroup.messageIds)
                 continue
             }
```

Comment:
```suggestion
            when {
                msgProtocols.none(String::isBlank) && protocols.none(msgProtocols::contains) -> {
                    logger.debug { "Messages with $msgProtocols protocols instead of $protocols are presented" }
                    messageBatch.addGroups(messageGroup)
                    continue
                }
                !msgProtocols.all(String::isBlank) && !(msgProtocols.none(String::isBlank) && protocols.any(msgProtocols::contains)) -> {
                    val info = "No messages of $protocols protocols or mixed empty and non-empty protocols are present"
                    parentEventId.onErrorEvent(info, messageGroup.messageIds)
                    messageBatch.addGroups(messageGroup.toErrorMessageGroup(IllegalStateException(info), protocols))
                    continue
                }
            }
```